src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.CSharp.Tests.cs

```diff
@@ -0,0 +1,488 @@
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = a.ToLower().CompareTo(b);
+        return r;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = StringComparer.CurrentCultureIgnoreCase.Compare(a, b);
+        return r;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 17, 8, 41, StringComparerRule, CompareToName);
+        }
+
+        [Fact]
+        public Task Diagnostic_CompareTo_ToLowerInvariant_Assign()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = a.ToLowerInvariant().CompareTo(b);
+        return r;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = StringComparer.InvariantCultureIgnoreCase.Compare(a, b);
+        return r;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 17, 8, 50, StringComparerRule, CompareToName);
+        }
+
+        [Fact]
+        public Task Diagnostic_CompareTo_ToUpper_Assign()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = a.ToUpper().CompareTo(b);
+        return r;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = StringComparer.CurrentCultureIgnoreCase.Compare(a, b);
+        return r;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 17, 8, 41, StringComparerRule, CompareToName);
+        }
+
+        [Fact]
+        public Task Diagnostic_CompareTo_ToUpperInvariant_Assign()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = a.ToUpperInvariant().CompareTo(b);
+        return r;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = StringComparer.InvariantCultureIgnoreCase.Compare(a, b);
+        return r;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 17, 8, 50, StringComparerRule, CompareToName);
+        }
+
+        private Task VerifyCSharpAsync(string originalSource, string fixedSource,
+            int startLine, int startColumn, int endLine, int endColumn, DiagnosticDescriptor rule, string argument)
+        {
+            DiagnosticResult diagnosticResult = VerifyCS
+                .Diagnostic(rule)
+                .WithArguments(argument)
+                .WithSpan(startLine, startColumn, endLine, endColumn);
```

Comment:
This overload should be avoided. Instead, the markup syntax should be used `[| |]` like this:

https://github.com/dotnet/roslyn-analyzers/blob/2b6ab8d727ce73a78bcbf026ac75ea8a7c804daf/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerationsTests.cs#L111

Reply:
Totally forgot about this pattern. I copied the pattern used in one of my old analyzers. I'll switch to the suggested one.

Reply:
Ok I switched, but I am unsure if I can still specify whifch rule to expect. Do you know if it's possible? I haven't been able to find code examples doing this.

Also, I had to use MarkupOptions.UseFirstDescriptor to be able to detect a single rule. Unsure if this is the best practice. Let me know if you know.

Reply:
If you want to assert the argument (as it is done currently), you can go with `{|#0:code|}` syntax instead. Example:

https://github.com/dotnet/roslyn-analyzers/blob/2b6ab8d727ce73a78bcbf026ac75ea8a7c804daf/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureTests.Enums.cs#L78-L111